astropy/stats/circstats.py

```diff
@@ -0,0 +1,555 @@
+    >>> circmean(data_rad) # doctest: +FLOAT_CMP
+    <Quantity 0.848704412445019 rad>
+    >>> circmean(data_deg, deg=True) # doctest: +FLOAT_CMP
+    <Quantity 48.62718088722989 deg>
+    >>> from astropy import units as u
+    >>> circmean(data_deg*u.deg, deg=True) # doctest: +FLOAT_CMP
+    <Quantity 48.62718088722989 deg>
+    >>> circmean(data_rad*u.rad) # doctest: +FLOAT_CMP
+    <Quantity 0.848704412445019 rad>
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
+    """
+    return _angle(data, deg, w, 1, 0.0, axis)
+
+
+def circvar(data, deg=False, w=None, axis=None):
+    """ Computes the circular variance of an array of circular data.
+
+    There are some concepts for defining measures of dispersion for circular
+    data. The variance implemented here is based on the definition given by [1]
+    , which is also the same used by the R package 'CircStats' [2].
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array of circular (directional) data.
+    deg : boolean, optional
+        If ``data`` is expressed in degrees, this should be ``True``.
+        If ``deg`` is ``False``, the data is assumed to be measured in radians.
+    w : numpy.ndarray, optional
+        In case of grouped data, the i-th element of ``w`` represents a
+        weighting factor for each group such that sum(w,axis) equals the number
+        of observations. See [1], remark 1.4, page 22, for detailed
+        explanation.
+    axis : int, optional
+        Axis along which circular variances are computed. The default is to
+        compute the variance of the flattened array.
+
+    Returns
+    -------
+    circvar : float
+        Circular variance.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.stats import circvar
+    >>> data_deg = np.array([51, 67, 40, 109, 31, 358])
+    >>> data_rad = np.deg2rad(data_deg)
+    >>> circvar(data_rad) # doctest: +FLOAT_CMP
+    <Quantity 0.16356352748437508>
+    >>> circvar(data_deg, deg=True) # doctest: +FLOAT_CMP
+    <Quantity 0.16356352748437508>
+    >>> from astropy import units as u
+    >>> circvar(data_rad*u.rad) # doctest: +FLOAT_CMP
+    <Quantity 0.16356352748437508>
+    >>> circvar(data_deg*u.deg, deg=True) # doctest: +FLOAT_CMP
+    <Quantity 0.16356352748437508>
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
```

Comment:
It would be good to add a `Note` about the difference between this and the circvar in scipy